datalad/distribution/create.py

```diff
@@ -157,61 +162,69 @@ def __call__(
-                annex_version=annex_version,
-                annex_backend=annex_backend,
-                git_opts=git_opts,
-                annex_opts=annex_opts,
-                annex_init_opts=annex_init_opts)
 
-        else:
-            if no_annex:
-                if description:
-                    raise ValueError("Incompatible arguments: cannot specify "
-                                     "description for annex repo and declaring "
-                                     "no annex repo.")
-                if annex_opts:
-                    raise ValueError("Incompatible arguments: cannot specify "
-                                     "options for annex and declaring no "
-                                     "annex repo.")
-                if annex_init_opts:
-                    raise ValueError("Incompatible arguments: cannot specify "
-                                     "options for annex init and declaring no "
-                                     "annex repo.")
-
-                lgr.info("Creating a new git repo at %s", ds.path)
-                vcs = GitRepo(ds.path, url=None, create=True,
-                              git_opts=git_opts)
+            if sds is not None:
+                return sds.create_subdataset(
+                    ds.path,
+                    name=name,
+                    description=description,
+                    no_annex=no_annex,
+                    annex_version=annex_version,
+                    annex_backend=annex_backend,
+                    git_opts=git_opts,
+                    annex_opts=annex_opts,
+                    annex_init_opts=annex_init_opts)
             else:
-                # always come with annex when created from scratch
-                lgr.info("Creating a new annex repo at %s", ds.path)
-                vcs = AnnexRepo(ds.path, url=None, create=True,
-                                backend=annex_backend,
-                                version=annex_version,
-                                description=description,
-                                git_opts=git_opts,
-                                annex_opts=annex_opts,
-                                annex_init_opts=annex_init_opts)
-                # record the annex uuid of this repo for this afterlife
-                # to be able to track siblings and children
-                if 'datalad.annex.origin' in ds.config:
-                    # make sure we reset this variable completely, in case of a re-create
-                    ds.config.unset('datalad.annex.origin', where='dataset')
-                ds.config.add('datalad.annex.origin', vcs.uuid, where='dataset')
-                ds.repo.add('.datalad', git=True)
-
-            vcs.commit(msg="datalad initial commit",
-                       options=to_options(allow_empty=True))
-            # reset ID, we have a VCS now
-            ds._id = None
-            return ds
+                if isinstance(add_to_super, bool):
```

Comment:
Well, how about `if add_to_super is True` instead of just `if
add_to_super`? ;-)
Anyways - do as you like it.

2016-09-02 14:37 GMT+02:00 Yaroslav Halchenko <notifications@github.com>:

> In datalad/distribution/create.py
> <https://github.com/datalad/datalad/pull/746#discussion_r77338747>:
>
> > -                                annex_opts=annex_opts,
> > -                                annex_init_opts=annex_init_opts)
> > -                # record the annex uuid of this repo for this afterlife
> > -                # to be able to track siblings and children
> > -                if 'datalad.annex.origin' in ds.config:
> > -                    # make sure we reset this variable completely, in case of a re-create
> > -                    ds.config.unset('datalad.annex.origin', where='dataset')
> > -                ds.config.add('datalad.annex.origin', vcs.uuid, where='dataset')
> > -                ds.repo.add('.datalad', git=True)
> > -
> > -            vcs.commit(msg="datalad initial commit",
> > -                       options=to_options(allow_empty=True))
> > -            # reset ID, we have a VCS now
> > -            ds._id = None
> > -            return ds
> > +                if isinstance(add_to_super, bool):
>
> it could be a bool or a string. so this one is checking after it was
> checked that it is some kind of non-False value and I need to be sure I am
> not catching some other string or anything else as "True"
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/datalad/datalad/pull/746/files/005f4a61c7a633c49f4350b695d5db1d38c9c128#r77338747>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKAw_RdmCBjqj7EqvotLZ-al3e5Eyav6ks5qmBicgaJpZM4JyQnT>
> .
>